CGAL_ImageIO/include/CGAL/IO/read_vtk_image_data.h

```diff
@@ -69,26 +69,55 @@ read_vtk_image_data(vtkImageData* vtk_image, Image_3::Own owning = Image_3::OWN_
   image->ty = static_cast<float>(offset[1]);
   image->tz = static_cast<float>(offset[2]);
   image->endianness = ::_getEndianness();
+
   int vtk_type = vtk_image->GetScalarType();
   if(vtk_type == VTK_SIGNED_CHAR) vtk_type = VTK_CHAR;
-  if(vtk_type < 0 || vtk_type > VTK_DOUBLE)
-    vtk_type = VTK_DOUBLE;
-  const VTK_to_ImageIO_type_mapper& imageio_type =
-    VTK_to_ImageIO_type[vtk_type];
+  if(vtk_type < 0 || vtk_type > VTK_DOUBLE) vtk_type = VTK_DOUBLE;
+  const VTK_to_ImageIO_type_mapper& imageio_type = VTK_to_ImageIO_type[vtk_type];
   image->wdim = imageio_type.wdim;
   image->wordKind = imageio_type.wordKind;
   image->sign = imageio_type.sign;
+
+  const int cn = vtk_image->GetNumberOfScalarComponents();
+
   if (!vtk_image->GetPointData() || !vtk_image->GetPointData()->GetScalars()) {
     ::_freeImage(image);
     return Image_3();
   }
+
+  // If there is more than a scalar per point, vtk_image->data is not immediately
+  // interpretable in Image_3->data
+  CGAL_assertion(owning == Image_3::OWN_THE_DATA || cn == 1);
+
   CGAL_assertion(vtk_image->GetPointData()->GetScalars()->GetNumberOfTuples() == dims[0]*dims[1]*dims[2]);
+
   if(owning == Image_3::OWN_THE_DATA) {
-    image->data = ::ImageIO_alloc(dims[0]*dims[1]*dims[2]*image->wdim);
-    // std::cerr << "GetNumberOfTuples()=" << vtk_image->GetPointData()->GetScalars()->GetNumberOfTuples()
-    //           << "\nimage->size()=" << dims[0]*dims[1]*dims[2]
-    //           << "\nwdim=" << image->wdim << '\n';
-    vtk_image->GetPointData()->GetScalars()->ExportToVoidPointer(image->data);
+    int dims_n = dims[0]*dims[1]*dims[2];
+    image->data = ::ImageIO_alloc(dims_n * image->wdim);
+    std::cerr << "GetNumberOfTuples() = " << vtk_image->GetPointData()->GetScalars()->GetNumberOfTuples() << "\n"
+              << "components = " << cn << "\n"
+              << "wdim = " << image->wdim << "\n"
+              << "image->size() = " << dims_n << std::endl;
+
+    if(cn == 1) {
+      vtk_image->GetPointData()->GetScalars()->ExportToVoidPointer(image->data);
+    } else {
+      std::cerr << "Warning: input has " << cn << " components; only the value of the first component will be used." << std::endl;
```

Comment:
Can we check that the components have the same value (in order to encode a grey value)